toolbox/fdc3-for-web/fdc3-web-impl/src/handlers/HeartbeatHandler.ts

```diff
@@ -95,35 +101,41 @@ export class HeartbeatHandler implements MessageHandler {
     clearInterval(this.timerFunction);
   }
 
-  accept(msg: any, sc: ServerContext<AppRegistration>, from: InstanceID): void {
+  accept(
+    msg: AppRequestMessage | WebConnectionProtocol6Goodbye,
+    sc: ServerContext<AppRegistration>,
+    from: InstanceID
+  ): void {
     if (!this.contexts.includes(sc)) {
       this.contexts.push(sc);
     }
 
     if (msg.type == 'heartbeatAcknowledgementRequest') {
       const app = sc.getInstanceDetails(from);
       if (app) {
-        this.lastHeartbeats.set(app.instanceId!!, new Date().getTime());
+        this.lastHeartbeats.set(app.instanceId, new Date().getTime());
       }
     }
 
-    if (msg.type == 'WCP5Shutdown') {
+    if (msg.type == 'WCP6Goodbye') {
       const app = sc.getInstanceDetails(from);
       if (app) {
         sc.setAppState(from, State.Terminated);
       }
     }
   }
 
-  async sendHeartbeat(sc: ServerContext<AppRegistration>, app: AppIdentifier): Promise<void> {
-    const heartbeatEventMessage: BrowserTypes.HeartbeatEvent = {
-      type: 'heartbeatEvent',
-      meta: {
-        timestamp: new Date(),
-        eventUuid: sc.createUUID(),
-      },
-      payload: {},
-    };
-    sc.post(heartbeatEventMessage, app.instanceId!!);
+  async sendHeartbeat(sc: ServerContext<AppRegistration>, app: FullAppIdentifier): Promise<void> {
+    sc.post(
```

Comment:
Could this look like `const event: HeartbeatEvent = { ... }; sc.post(event);`?

Reply:
yup